akka-http2-support/src/main/scala/akka/http/impl/engine/http2/RequestParsing.scala

```diff
@@ -66,8 +68,12 @@ private[http2] object RequestParsing {
           rec(remainingHeaders.tail, method, scheme, authority, path, contentType, contentLength, headers)
 
         case (name, value) ⇒
-          // FIXME: parse to real headers
-          rec(remainingHeaders.tail, method, scheme, authority, path, contentType, contentLength, headers += RawHeader(name, value))
+          // FIXME: later modify by adding HttpHeaderParser.parseHttp2Header that would use (name, value) pair directly
+          //        or use a separate, simpler, parser for Http2
+          val concHeaderLine = name + ": " + value + "\r\nx"
+//          println("Parsing header line: " + concHeaderLine)
+          httpHeaderParser.parseHeaderLine(ByteString(concHeaderLine))()
```

Comment:
Continuing, maybe as for HTTP2 instead of long string containing all headers we have nice (name, value) pairs, we could add method `parseNameValuePair` taking `(String, String)` or add a separate caching parser for HTTP2.
WDYT?